pkg/puzzles/puzzles.go

```diff
@@ -249,13 +248,37 @@ func answersAreEqual(userAnswer *ipc.ClientGameplayEvent, correctAnswer *macondo
 	}
 
 	return converted.Type == correctAnswer.Type &&
-		converted.Row == correctAnswer.Row &&
-		converted.Column == correctAnswer.Column &&
-		converted.Direction == correctAnswer.Direction &&
+		positionsAreEqual(converted, correctAnswer) &&
 		converted.PlayedTiles == correctAnswer.PlayedTiles &&
 		utilities.SortString(converted.Exchanged) == utilities.SortString(correctAnswer.Exchanged)
 }
 
+func positionsAreEqual(userAnswer *macondopb.GameEvent, correctAnswer *macondopb.GameEvent) bool {
+	return (userAnswer.Row == correctAnswer.Row &&
+		userAnswer.Column == correctAnswer.Column &&
+		userAnswer.Direction == correctAnswer.Direction) ||
+		(answerHitsCenterSquare(correctAnswer) &&
+			userAnswer.Row == correctAnswer.Column &&
+			userAnswer.Column == correctAnswer.Row &&
+			userAnswer.Direction != correctAnswer.Direction)
+}
+
+func answerHitsCenterSquare(answer *macondopb.GameEvent) bool {
+	var startSquare int32
+	if answer.Row == 7 && answer.Column <= 7 && answer.Direction == macondopb.GameEvent_HORIZONTAL {
```

Comment:
if we have ZOMGWords puzzles we'll have to not hardcode these numbers. I don't see a straightforward way to pass board dimensions all the way down to here though. It's probably ok for now.